src/esp/physics/bullet/BulletURDFImporter.cpp

```diff
@@ -646,9 +654,17 @@ Mn::Matrix4 BulletURDFImporter::ConvertURDF2BulletInternal(
       if (collisionFlags & io::URDF::HAS_COLLISION_GROUP) {
         collisionFilterGroup = colGroup;
       }
+
+      int collisionFilterMask = CollisionGroupHelper::getMaskForGroup(
+          CollisionGroup(collisionFilterGroup));
+// We don't like overriding the mask in the URDF; we disable support for this.
+// We prefer to only override the group, while still using getMaskForGroup
+// (above) for mask computation.
+#if 0
       if (collisionFlags & io::URDF::HAS_COLLISION_MASK) {
         collisionFilterMask = colMask;
       }
+#endif
```

Comment:
I can envision experimental situations where one might want to manipulate the potential collisions of a single linkage in an explicit manner without changing its group membership.  I think having the mask override being supported in the xml would be useful and much less arcane than determining the proper group that has the desired target group membership.  How this mask input is consumed could perhaps be decided by another collision tag-level input (say "mask_type" : additive, subtractive, override or something like that)

Reply:
First, beware that the URDF importer always seems to set both HAS_COLLISION_GROUP and HAS_COLLISION_MASK, even if you've only specified one in the URDF. So that would need to change if we're considering re-enabling this code here.

>much less arcane than determining the proper group
I don't consider this arcane. The group is conceptual and usually well-named.

Reply:
>I can envision experimental situations where one might want to manipulate the potential collisions of a single linkage in an explicit manner without changing its group membership. 

Can you give an explicit example?

Reply:
> I don't consider this arcane. The group is conceptual and usually well-named.
The names of the groups don't reflect what collisions they allow.  They also don't come close to covering all the possible masking groups. For instance, using the example above, what if i want to record kinematic-kinematic collision point locations? Is there a group that allows for all but static + noncollidable?  Kinematic may not require a response, but the information for the location could be very useful.

> Can you give an explicit example?
Record collision point locations of a kinematically-posed linkage component to see if/where it is hitting the stage.    


I guess my point in bringing this up is that we can hardcode masks to 32 common, common-sense, collision groups describing the majority of feasible behavior, but we should design around supporting research-driven situations where "uncommon" behavior may be desirable, particularly for diagnostic purposes.  

Why would making this functionality available to the user be undesirable?

Reply:
This is probably better discussed in-person just for time's sake. I am just giving my two cents here; ultimately @aclegg3  owns the physics API. Also, the AO branch is still just a branch so what we merge here doesn't represent a final decision.